src/sinks/aws_cloudwatch_logs.rs

```diff
@@ -227,47 +333,93 @@ impl Service<Vec<Event>> for CloudwatchLogsSvc {
     }
 }
 
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+struct CloudwatchKey {
+    group: Bytes,
+    stream: Bytes,
+}
+
+fn interpolate(s: &str) -> Partition {
+    use regex::Regex;
+    let r = Regex::new(r"\{event\.(?P<key>\D+)\}").unwrap();
```

Comment:
If I'm understanding this correctly, an input string like `event-logs-#{event.application}-2` with an event where `application` is `foo` will result in a key of just `foo`. Right?

I see you added one happy path test, but this is probably something that needs quite a bit more than that. We should also be clear what kind of syntax we're willing to support here.